src/test/groovy/org/prebid/server/functional/tests/BidderParamsSpec.groovy

```diff
@@ -385,4 +386,36 @@ class BidderParamsSpec extends BaseSpec {
         assert bidderRequest.ext.prebid.server.datacenter == serverDataCenter
         assert bidderRequest.ext.prebid.server.gvlId == serverHostVendorId
     }
+
+    def "PBS should send request to bidder with header Content-Encoding = gzip when adapters.BIDDER.endpointCompression = GZIP"() {
+        given: "PBS with adapter configuration"
+        def pbsService = pbsServiceFactory.getService(["adapters.generic.enabled"               : "true",
+                                                        "adapters.generic.endpointCompression"   : CompressionType.GZIP.name()])
+
+        and: "Default basic generic BidRequest"
+        def bidRequest = BidRequest.defaultBidRequest
+
+        when: "PBS processes auction request"
+        def response = pbsService.sendAuctionRequest(bidRequest)
+
+        then: "Response should contain header Content-Encoding = gzip"
+        assert response.ext.debug.httpcalls[BidderName.GENERIC.value]
```

Comment:
1. 'response' can't be null, '?.' just after it may be omitted
2. accessing value with '[]' operator will throw an NPE if current data structure is null (or some null was even before an accessed data structure). To guarantee a null-safety get() method should be called (like httpcalls?.get(BidderName.GENERIC.value))